pkg/apis/rook.io/v1alpha2/network.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2019 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package v1alpha2
+
+import (
+	"encoding/json"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func IsHostNetworkDefined(net NetworkSpec) bool {
+	return net.Provider == "host"
+}
+
+func IsMultusNetworkDefined(net NetworkSpec) bool {
+	return !IsHostNetworkDefined(net) && net.Provider == "multus"
+}
+
+func parseMultusSelector(ns NetworkSelector) map[string]string {
+	rawMap := make(map[string]string)
+
+	err := json.Unmarshal([]byte(ns), &rawMap)
+
+	if err != nil {
+		// it can be in short form
+		multusSelectorString := string(ns)
+
+		nsEndIndex := strings.IndexAny(multusSelectorString, "/")
+		if nsEndIndex != -1 {
+			rawMap["namespace"] = multusSelectorString[:nsEndIndex]
+		}
+
+		ifStartIndex := strings.LastIndexAny(multusSelectorString, "@")
+		if ifStartIndex != -1 && len(multusSelectorString)-ifStartIndex > 1 {
+			rawMap["interface"] = multusSelectorString[ifStartIndex+1:]
+		}
+
+		if nsEndIndex != -1 && ifStartIndex != -1 && ifStartIndex-nsEndIndex > 1 {
+			rawMap["name"] = multusSelectorString[nsEndIndex+1 : ifStartIndex]
+		} else if nsEndIndex == -1 && ifStartIndex != -1 {
+			rawMap["name"] = multusSelectorString[:ifStartIndex]
+		} else if nsEndIndex != -1 && ifStartIndex == -1 {
+			rawMap["name"] = multusSelectorString[nsEndIndex+1:]
+		}
+	}
+
+	return rawMap
+}
+
+func GetMultusIfName(ns NetworkSelector) string {
+	multusMap := parseMultusSelector(ns)
+	ifName := "net1"
+
+	if name, ok := multusMap["interfaceRequest"]; ok {
+		ifName = name
+	}
+	if name, ok := multusMap["interface"]; ok {
+		ifName = name
+	}
+
+	return ifName
+}
+
+// ApplyMultus apply multus selector to Pods
+// Multus supports short and json syntax, use only one kind at a time.
+func ApplyMultus(net NetworkSpec, objectMeta *metav1.ObjectMeta) {
```

Comment:
To apply properties to pods, this method would be better in the `pkg/operator/k8sutil` package. You could either add it to pod.go, or maybe to a new network.go. Other methods in this file might make sense as well to move.

Reply:
Ok, I leave the method form functions in `apis/` but moved other functions to k8sutil.